### ISSUE TITLE
EPX upscaler is broken

### ISSUE THREAD

**Issue by dirkwhoffmann** [dirkwhoffmann/vAmiga#302]
😯

<img width="951" alt="Bildschirmfoto 2020-03-23 um 19 14 45" src="https://user-images.githubusercontent.com/12561945/77349071-c2166600-6d3a-11ea-9ddb-85bbcc2cc254.png">
<img width="643" alt="Bildschirmfoto 2020-03-23 um 19 15 06" src="https://user-images.githubusercontent.com/12561945/77349080-c3e02980-6d3a-11ea-95be-ae6966148488.png">

**Comment by dirkwhoffmann** [comment]
There is no fix yet, but at least I know why it is broken: The upscaling pipeline has two stages. The first stage invokes an in-texture upscaler that keeps the texture size unchanged. The second stage invokes a standard upscaler that doubles the texture size. The first stage is meant for lores mode only, because one lores pixel is two texture pixels. Hence, to get things right, the GPU needs to know if a particular line was drawn in lores mode or hires mode by the Amiga. The old implementation encoded this information in the last pixel (kind of a hidden watermark). Unfortunately, this doesn't work any more, because the texture cutout is no longer fixed (it gets stretched and moved by the video setting controls).

**Comment by dirkwhoffmann** [comment]
I fooled myself. The texture encoding trick still works.

The red color is debug output which I forgot to remove.  A couple of month ago I experimented with smart code that was supposed to automatically detect lores / hires areas. The red color marks the areas that have been discovered to be hires. It's like the red pill given by dentists to exhibit the areas on your teeth and gum that weren't brushed properly.
